core/domain/takeout_service.py

```diff
@@ -42,34 +45,27 @@ def export_data_for_user(user_id):
                                 model export policy>
         }
     """
-    models_to_export = [
-        user_models.UserStatsModel,
-        user_models.UserSettingsModel,
-        user_models.UserSubscriptionsModel,
-        user_models.UserSkillMasteryModel,
-        user_models.UserContributionsModel,
-        user_models.ExplorationUserDataModel,
-        user_models.CompletedActivitiesModel,
-        user_models.IncompleteActivitiesModel,
-        user_models.ExpUserLastPlaythroughModel,
-        user_models.LearnerPlaylistModel,
-        user_models.CollectionProgressModel,
-        user_models.StoryProgressModel,
-        feedback_models.GeneralFeedbackThreadModel,
-        feedback_models.GeneralFeedbackMessageModel,
-        collection_models.CollectionRightsModel,
-        suggestion_models.GeneralSuggestionModel,
-        exploration_models.ExplorationRightsModel,
-        email_models.GeneralFeedbackEmailReplyToIdModel
+    all_models = []
+    model_names_list = [
```

Comment:
Actually, this seems like the wrong place to do this sort of check. The check should be done in a test, similar to how wipeout does it. Otherwise the first time we'll know about an error is when it happens on the user side.

I think here, it makes sense to have an explicit list of models -- maybe have a method in this class that exposes this list, and make use of that method here. Then there should be a backend test to ensure that all models are accounted for and the correct ones are in this list (any others should have the correctly-specified takeout policy that exempts them from this list).

Reply:
Changed in most recent commits.